torchsynth/synth.py

```diff
@@ -252,12 +252,12 @@ def __init__(self, synthglobals: SynthGlobals, *args, **kwargs):
         self.add_synth_modules(
             [
                 ("keyboard", MonophonicKeyboard),
-                ("adsr_1", ADSR),
-                ("adsr_2", ADSR),
-                ("lfo_1", LFO),
-                ("lfo_2", LFO),
-                ("lfo_1_amp_adsr", ADSR),
-                ("lfo_2_amp_adsr", ADSR),
+                ("adsr_1", ADSR, {"upsample": True}),
+                ("adsr_2", ADSR, {"upsample": True}),
+                ("lfo_1", LFO, {"upsample": True}),
+                ("lfo_2", LFO, {"upsample": True}),
+                ("lfo_1_amp_adsr", ADSR, {"upsample": True}),
+                ("lfo_2_amp_adsr", ADSR, {"upsample": True}),
```

Comment:
Now in the synth we just need the modulation modules that output to audio rate modules to upsample their output. Other modules like the adsrs for the LFOs output at control rate.

Reply:
Can we be explicit about when upsampling does or doesn't happen and make it required?

The default scares me a little

Reply:
Yes let's drop the default.